api/src/main/java/me/lokka30/treasury/api/common/services/Service.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/lokka30/Treasury>.
+ */
+
+package me.lokka30.treasury.api.common.services;
+
+import java.util.Objects;
+import java.util.function.Supplier;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Represents a service.
+ *
+ * @param <T> type of the data held
+ * @author MrIvanPlays
+ */
+public final class Service<T> implements Comparable<Service<?>>, Supplier<T> {
+
+    private final String registrator;
+    private final ServicePriority priority;
+    private final T service;
+
+    public Service(
+            @NotNull String registrator, @NotNull ServicePriority priority, @NotNull T service
```

Comment:
`service` should probably be `serviceType`, also `Service get()` should probably be `getServiceType()` if you agreed.

Reply:
not really. we're providing a service, not a service type. can't agree for both of these things here.

Reply:
Ok. But I think `get` should be `getService` because when combined with optionals it looks like `get().get()`. Nitpick though, feel free to mark this as resolved.